src/loader.c

```diff
@@ -29,9 +29,29 @@
 #include <dlfcn.h>
 #include <stdio.h>
 #include <stdlib.h>
+#include <string.h>
 
 static void* s_impl_library = 0;
 static struct wpe_loader_interface* s_impl_loader = 0;
+#ifndef WPE_BACKEND
+static char* s_impl_library_name;
+#endif
+
+static void
+wpe_loader_set_library_name(const char* library_name)
+{
+    size_t len;
+
+    if (!library_name)
+        return;
+
+    len = strlen(library_name) + 1;
+    if (len == 1)
+        return;
+
+    s_impl_library_name = (char *)malloc(len);
```

Comment:
Preferably we would avoid using malloc, and have s_impl_library_name be a 256- or 512-byte char array that we copy the name into. One problem with that might be using path to the library as the argument here, which could run over the 512 byte limit. WDYT?

Reply:
I thought about it, but in the end I thought this was a one time malloc of just a few bytes in most of the cases, so it's probably ok. We could have a 512 char array and a pointer and only malloc if the  argument passed doesn't fit in the char array, for example.

Reply:
You can use `PATH_MAX` (4096)